cpp/src/text/normalize.cu

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_view.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/strings/strings_column_view.hpp>
+#include <cudf/strings/string_view.cuh>
+#include <strings/utilities.cuh>
+#include <strings/utilities.hpp>
+
+#include <nvtext/normalize.hpp>
+#include <text/utilities/tokenize_ops.cuh>
+
+#include <thrust/for_each.h>
+
+namespace nvtext
+{
+namespace detail
+{
+namespace
+{
+
+/**
+ * @brief Normalize spaces in a strings column.
+ *
+ * Repeated whitespace is replaced with a single space.
+ * Also, whitespace is trimmed from the beginning and end of each string.
+ *
+ * This functor can be called to compute the output size in bytes
+ * of each string and then called again to fill in the allocated buffer.
+ */
+struct normalize_spaces_fn
+{
+    cudf::column_device_view const d_strings;   // strings to normalize
+    int32_t const* d_offsets{}; // offsets into d_buffer
```

Comment:
`size_type` is incorrect. The offsets column is of type `INT32`

Reply:
This would be incorrect. If `size_type` is changed, then this would cause a failure. If offsets is changed to use `INT64` then this would change to use `int64_t` which would be more correct rather than rely on `size_type`. There is not column type that matches `size_type` primitive.